state/offerconnections.go

```diff
@@ -191,6 +191,23 @@ func (st *State) OfferConnectionForRelation(relationKey string) (*OfferConnectio
 	return newOfferConnection(st, &connDoc), nil
 }
 
+// OfferConnectionsForUser returns the offer connections for the specified user.
+func (st *State) OfferConnectionsForUser(username string) ([]*OfferConnection, error) {
+	offerConnectionCollection, closer := st.db().GetCollection(offerConnectionsC)
+	defer closer()
+
+	var connDocs []offerConnectionDoc
+	err := offerConnectionCollection.Find(bson.D{{"username", username}}).All(&connDocs)
+	if err != nil {
+		return nil, errors.Annotatef(err, "cannot get offer connection details for user %q", username)
+	}
+	conns := make([]*OfferConnection, len(connDocs))
+	for i, oc := range connDocs {
+		conns[i] = newOfferConnection(st, &oc)
+	}
+	return conns, nil
```

Comment:
This doesn't appear to support the notion of "allow everyone to access this" or even "allow a group to access this", but only supports explicit grants to specific people. Is that correct?

Reply:
everyone@external is a valid ACL user. But this offer connection entity is not the ACL, it is simply used to record a connection. The standard ACL model is used for permissions.